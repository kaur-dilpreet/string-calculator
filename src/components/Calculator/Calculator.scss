.calculator-output {
    padding: 20px 0;
}