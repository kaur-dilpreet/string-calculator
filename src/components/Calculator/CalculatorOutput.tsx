import { JSX } from "react";

interface ICalculatorOutput {
    result: number;
    error: string;
}
const CalculatorOutput = ({ result, error }: ICalculatorOutput): JSX.Element => {
    return (
        <div className="calculator-output">
            {error ? (
                <div style={{ color: 'red' }}>
                    Error: {error}
                </div>
            ) : (
                <div>
                    Result: {result}
                </div>
            )}
        </div>
    );
};

export default CalculatorOutput;