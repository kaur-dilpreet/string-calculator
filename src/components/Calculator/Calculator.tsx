import { JSX, useState } from "react";
import "./Calculator.scss";
import Input from "../shared/Input/Input";
import Button from "../shared/Button/Button";
import Header from "../shared/Header/Header";
import Footer from "../shared/Footer/Footer";
import CalculatorOutput from "./CalculatorOutput";

const Calculator = (): JSX.Element => {
    const [inputValue, setInputValue] = useState("");
    const [result, setResult] = useState<number>(0);
    const [error, setError] = useState<string>("");

    const add = (numbers: string): number => {
        if (numbers === "") return 0;

        // Case 1: If the input string contains a single number
        return parseInt(numbers, 10);
    };

    const handleSubmit = (): void => {
        try {
            const sum = add(inputValue);
            setResult(sum);
        } catch (e) {
            if (e instanceof Error) {
                setResult(0);
                setError(e.message);
            } else {
                setResult(0);
                setError('An unknown error occurred.');
            }
        }
    };

    return (
        <div className="app-main">
            <Header />
            <div className="app-content app-container">
                <h1>String Calculator</h1>
                <form
                    onSubmit={(e) => {
                        e.preventDefault();
                        handleSubmit();
                    }}
                    style={{ display: "flex", gap: "10px", alignItems: "center" }}
                >
                    <Input value={inputValue} onChange={setInputValue} placeholder="Enter string value" />
                    <Button onClick={handleSubmit} text="Submit" />
                </form>
                <CalculatorOutput result={result} error={error} />
            </div>
            <Footer />
        </div>
    );
};

export default Calculator;