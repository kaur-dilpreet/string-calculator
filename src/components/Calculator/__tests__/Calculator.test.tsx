import { shallow } from 'enzyme';
import Calculator from "../Calculator";

describe('Calculator Component', () => {
  it('should render the component correctly', () => {
    const wrapper = shallow(<Calculator />);
    expect(wrapper).toMatchSnapshot();
  });

  it('sum for empty value', () => {
    const wrapper = shallow(<Calculator />);

    const inputWrapper = wrapper.find('Input').dive();
    const input = inputWrapper.find('input');
    input.simulate('change', { target: { value: '' } });

    const buttonWrapper = wrapper.find('Button').dive();
    const button = buttonWrapper.find('button');
    button.simulate('click');

    const calculatorOutputWrapper = wrapper.find('CalculatorOutput').dive();
    expect(calculatorOutputWrapper.find('.calculator-output div').text()).toContain('Result: 0');
  });

  it('sum for single number', () => {
    const wrapper = shallow(<Calculator />);

    const inputWrapper = wrapper.find('Input').dive();
    const input = inputWrapper.find('input');
    input.simulate('change', { target: { value: '2' } });

    const buttonWrapper = wrapper.find('Button').dive();
    const button = buttonWrapper.find('button');
    button.simulate('click');

    const calculatorOutputWrapper = wrapper.find('CalculatorOutput').dive();
    expect(calculatorOutputWrapper.find('.calculator-output div').text()).toContain('Result: 2');
  });
});
