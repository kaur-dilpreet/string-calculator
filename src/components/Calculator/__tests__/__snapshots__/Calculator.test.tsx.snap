// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator Component should render the component correctly 1`] = `ShallowWrapper {}`;
